examples/mobile/HomeApp/js/app.js

```diff
@@ -224,17 +309,18 @@ import Storage from "./clipping-storage.js";
 
 		//TODO: all webclips url should checked if they are valid
 
-		if (!appsList.length) {
-			getAppsList.then((apps) => {
-				appsList = apps;
-				updateWebClipsUI();
-				updateWebClipListPopup();
-				storage.refreshStorage(Storage.elements.APPSLIST, apps);
-			});
-		} else {
+		if (appsList.length) {
 			updateWebClipsUI();
```

Comment:
those two functions will be called again later,
do we really want to call it twice?

Reply:
This code is twice but first code is synchronous and second code is asynchronous and may be called much later.